### ISSUE TITLE
arc() function docs say it only uses radians, but it actually listens to angleMode()

### ISSUE THREAD

**Issue by worthant** [processing/p5.js#7562]
### Most appropriate sub-area of p5.js?

- [x] Accessibility
- [ ] Color
- [x] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [x] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [ ] Friendly errors
- [ ] Other (specify if possible)

### p5.js version

1.11.3

### Web browser and version

all of them

### Operating system

Linux Fedora Workstation 39 && Windows 11

### Steps:
1. Open web editor
2. Write simple code to create an arc
3. See that it's indeed degrees, not radians

### Snippet:

```js
function setup() {
  createCanvas(400, 400);
}

function draw() {
  background(220);
  arc(150, 150, 50, 50, 0, 45)
}
```
> as you see, we draw an arc from 0 to 45 degrees, while it should be radians

here is what you get from this code:

|![Image](https://github.com/user-attachments/assets/690944fd-59d4-4c7e-b57b-35dfd8e132de)|
|-|

**Comment by davepagurek** [comment]
It looks like maybe this is actually using radians, but 45 radians wraps around enough times that it looks similar to a 45 degree angle. To test this, I manually am drawing a line along `y = x`, which is definitely 45 degrees, and it's not quite the same as the arc:

![Image](https://github.com/user-attachments/assets/133c2a71-8ccc-4fb9-83d5-187cea33a1db)

In the code, it looks like we are using `_toRadians`, which converts to radians if the `angleMode` is degrees and keeps it the same otherwise. So although the docs say the angles are always radians, it looks like it does actually respond correctly to degrees if you specify `angleMode(DEGREES)` beforehand, so an update to the docs here might be helpful.

https://github.com/processing/p5.js/blob/3c9de136e36912d6ced8b71a81e7413f3eab8b61/src/core/shape/2d_primitives.js#L327-L328

**Comment by worthant** [comment]
@davepagurek Hi!)

Thanks for your answer 👍
I forgot to mention this in the issue forms, but yes, in my own project i have `angleMode` as `DEGREES`, so the `arc()` function was interpreting radians as degrees, which was messing up with my whole math module. 

Thanks for pointing that out, that's actually my skill issue, because i implemented my own `sin`, `cos`, `tan`, `ctg` and other functions using Taylor series, and all my functions accept radians, as an input, while the mode is `DEGREES` for simplicity of drawing from the start.

I indeed think that it's necessary to mention this in the docs, as it's a bit confusing :) 

To clarify, 45 radians = 2578.31 grad
It just wrapped around 7 times, and then drew ~58 grad angle, so *your assumption of the issue itself was correct*.

![Image](https://github.com/user-attachments/assets/61380002-e155-41e7-a406-01b850adff88)

**Comment by worthant** [comment]
So, can i change p5.js docs now? Should i change them only for arc, or also for other functions, as i suppose this could be the thing for some other math functions?

**Comment by davepagurek** [comment]
That would be great! So to complicate things a little bit: we currently are close to a release for p5 2.0, and development for that is in the `dev-2.0` branch. We'd like to fix this forward for 2.0, but also it would be good to  keep the 1.x docs up to date too. So potentially we could have two PRs, one for each branch?

For other functions: if you notice other cases, that would also be great to fix! Ideally in most cases it supports both radians and degrees based on the mode, but if some existing docs specify just radians, we should do a quick check to see if it does actually support both before we assume it does, and hold off updating those docs for now. (And maybe file a separate issue to make it correctly handle angleMode for consistency 🙂)

**Comment by davepagurek** [comment]
I just updated the issue title for clarity!

**Comment by worthant** [comment]
@davepagurek Ok let's go like this: I'll create fixes for a dev-2.0 branch, and only when it will be approved i will just cherry-pick all the commits to the similar PR for the 1.x branch, because other way we could potentially have 1 million change requests in both of them (just looking in the future), which wouldn't be good for clarity.

How fast do you want it to be done? I'm working+studying currently, so is 1-3 days of delay acceptable?

**Comment by davepagurek** [comment]
Sounds good! As a heads up, I'm not certain how well cherry picking will work because we've moved some things around in the codebase, so there may be some manual merge conflicts to resolve after cherry picking -- let me know when you get there if you need any help with that.



> How fast do you want it to be done? I'm working+studying currently, so is 1-3 days of delay acceptable?

For sure! 2.0 won't get released until end of March anyway haha

**Comment by VANSH3104** [comment]
@davepagurek can i give it a try as this issue is assigned two weeks ago can you assign this to me ?

**Comment by VANSH3104** [comment]
@davepagurek  can you check my pr is anything missing or it is correct
